tensorflow/python/ops/script_ops.py

```diff
@@ -129,7 +129,108 @@ def __del__(self):
     _py_funcs.remove(self._token)
 
 
-def py_func(func, inp, Tout, stateful=True, name=None):
+def py_func(func, args=(), kwargs={}, output_types=None, output_shapes=None,
```

Comment:
This change from imp to args, kwargs can break existing code, breaking our API compatibility guidelines we've had since tf 1.0.

So we need to find a way of doing what you want while not changing the signature of py_func.

Reply:
When my suggestion does not move to contrib or 3-party and it is necessary to keep the API, my suggestion is (I do not know the rules for deprecated arguments in tf, I saw only some examples):
`def py_func(func, inp, Tout=None, stateful=True, name=None, kwargs={}, Sout=None)`

What I want with the changed signature is to allow keyword arguments and automatic set_shape like in tf.data.Dataset.from_generator.

Since in python it is common to use `func(*args, **kwargs)` I used that names. output_types and shape came from tf.data.Dataset.from_generator.